src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -717,5 +704,10 @@ shard_name(PG_FUNCTION_ARGS)
 	}
 
 	AppendShardIdToName(&relationName, shardId);
-	PG_RETURN_TEXT_P(cstring_to_text(relationName));
+
+	schemaId = get_rel_namespace(relationId);
```

Comment:
I see some duplication with `ConstructQualifiedShardName()`. Do we have any special reason?